### PR TITLE
Add ability to open Source assets in the workspace

### PR DESCRIPTION
**Pull request by ericpgreen2**
Addresses #345

@bcolloran, in reviewing this, if you think this is a meaningful improvement over what we currently have, then let's merge :) If you have any additional thoughts on what could make it better (I have a few of my own), let's open in a new issue just so we can move this forward incrementally.

### CODE REVIEW

File: src/lib/components/column-profile/CollapsibleTableHeader.svelte
Comment (bcolloran) [inline src/lib/components/column-profile/CollapsibleTableHeader.svelte:108]: see similar comment below about pulling business logic out of rendering code -- in a case like this, once we get up to 3 bools, we now have 8 possible boolean states that we'd need to check on if we wrote a test around this rendering code. That much rendering logic is a lot to test, b/c testing rendering sucks -- needs all kinds of harnesses and mocks and stuff.

On the other hand, we could roll these three booleans into a single boolean  `showEntityDetails = titleElementHovered || emphasizeTitle || contextMenuOpen`  in the `<script>` section of this component, in which case there are just two UI states to test, and the 8 states of the boolean can be tested in plain code.

This is kind of a silly example, because you probably wouldn't test the states of and 8 state boolean that is defined like this, since the test would basically match the defintion. And this might seem a bit pedantic, because... yes, I know: we have almost no UI component tests today. So I'm not actually asking you to change this code in this particular instance :-) I just noticed it as a pattern that I see a lot in the codebase (a lot of logic in presentational code), and I'm starting to look down the road to the day when hopefully we _do start testing the UI_, so I thought I'd point out this pattern and suggest an option that a lot of people use to simplify UI testing. It's just a thought, for whatever that's worth, :-)

**Review by bcolloran — CHANGES_REQUESTED** [review]
Love this functionality @ericpgreen2!

I noticed a small regression in the tooltip click functionality, and I think there might be opportunities to tighten up some of the surrounding code a bit.

File: src/lib/components/column-profile/NavEntry.svelte
Comment (ericpgreen2) [reply src/lib/components/column-profile/NavEntry.svelte]: Nice, made those changes, looks cleaner 👍

File: src/lib/util/command-click-action.ts
Comment (ericpgreen2) [reply src/lib/util/command-click-action.ts:1]: I tracked down the animation bug – it's due to the `rill:app:ui:[key]-click-action-callbacks` being defined (`setContext`) and accessed (`getContext`) in sibling components, not in parent-child components. It was actually introduced in #405, so I'll open up a new PR to address it – to avoid conflating concerns and to keep this PR smaller.

File: src/routes/_surfaces/assets/TableAssets.svelte
Comment (ericpgreen2) [reply src/routes/_surfaces/assets/TableAssets.svelte:94]: I spotted that too – the warning is due to `addModel()` in `data-modeler-service/ModelActions.ts` having two possible return types (`ActionResponse` or `PersistentModelEntity`). Not ideal. But I considered it out-of-scope of this PR to address that ambiguity.

File: src/lib/components/tooltip/StackingWord.svelte
Comment (ericpgreen2) [reply src/lib/components/tooltip/StackingWord.svelte:12]: I very much agree that this isn't very elegant, but refactoring these actions to use one context is a bit of a rabbit hole, and I made the conscious decision to incur some tech debt here to get this done quickly. I'd vote for doing that refactor another time.

Comment (bcolloran) [reply src/lib/components/tooltip/StackingWord.svelte:12]: I don't think this is enough of a big deal to block on it, so I'm going to go ahead and mark this PR approved @ericpgreen2. But I would still recommend spending the cycles to avoid adding extra technical debt at this time. First, I don't think it will be that much of a rabbit hole; second, yes we are looking towards the July release, but I don't think we're in such dire time crunch yet that we can't spend cycles on code quality as we go; and third I'll offer my general perspective, which you can take for whatever it's worth, and even if you don't want to follow it now (which would be fine), maybe it's worth saying for later :-)

(I'm sure you've heard this perspective from other people before, because it's a boring and commonplace perspective; and I expect you will have guessed that this is my approach by now, but since we're still getting to know each other's working styles, I thought I'd go ahead an write it out :-) )

Little bits of tech debt like this begin accruing interest as soon as they are merged, and this issue you're working on and it's surrounding code will probably never be as clear to you as they are right now when you are fresh on this part of the code -- and they will certainly be less clear to someone else in 3 months when they look at this code and wonder: "oh geez, why was this nearly identical code copy/pasted? why are there two contexts? was there some subtle nuance in here that required that approach? some state that absolutely must be managed separately? some side effect that each of these units of copy/pasted has on the rest of the system that prevents them being abstracted? is this even safe to touch?" So, agreed, it may be a bit of extra work today... but it may be a 1.5x extra work for you later, or 3x work for someone else later.

Comment (ericpgreen2) [reply src/lib/components/tooltip/StackingWord.svelte:14]: That's all fair :). I just added a comment in `StackingWord.svelte` to clarify that it's tech debt, so we don't take the decision as gospel.

**Pull request merged**
